### DISCUSSION TITLE
Why does the merge randomly pick different values during different runs for the exact same order of updates?

### DISCUSSION THREAD

**Discussion by vineetdigit** [automerge/automerge#553] [Q&A] [answered]
If you run following snippet of code multiple times, the output is (randomly) either 'acme-our' or 'acme-their'

Why does it not reapeatedly resolve the key to the same value, even though nothing is really changing with the order  of updates?

```
const Automerge = require("@automerge/automerge");

let originDoc = Automerge.init();
originDoc = Automerge.change(originDoc, "set name", (doc) => {
  doc.key = "acme";
});

let ourDoc = Automerge.init();
ourDoc = Automerge.merge(ourDoc, originDoc);
ourDoc = Automerge.change(ourDoc, "set name", (doc) => {
  doc.key = "acme-our";
});

let theirDoc = Automerge.init();
theirDoc = Automerge.merge(theirDoc, originDoc);
theirDoc = Automerge.change(theirDoc, "set name", (doc) => {
  doc.key = "acme-their";
});

let mergedDoc = Automerge.init();
mergedDoc = Automerge.merge(mergedDoc, originDoc);
mergedDoc = Automerge.merge(mergedDoc, ourDoc);
mergedDoc = Automerge.merge(mergedDoc, theirDoc);

console.log(mergedDoc.key); // 'acme-our' or 'acme-their'?

```

**Comment by heckj** [comment, marked as answer]
This is almost exactly covered in the [Automerge documentation on managing conflicts](https://automerge.org/docs/cookbook/conflicts/). The reason it's appearing as random is because without any history to know which change came first, those changes are in direct conflict with each other - and the ordering of the result boils down to the ordering of the unique identifier used for ActorID. If you use [Automerge.getConflicts](https://automerge.org/docs/tsapi/namespaces/unstable/#getconflicts), you'll see the detail exposed there.

To avoid this scenario, create a single document and clone it using [Automerge.clone](https://automerge.org/docs/tsapi/namespaces/unstable/#clone) - the the documents can use the ordering of the changes to make a more deterministic result for you.

**Reply by ppoulard** [reply]
You can also give a specific actor ID to you 4 documents :
```
let originDoc = Automerge.init('0000000000000000000000000badcafe');
// use different actor IDs for the other docs
```
...and you will get the same output.
(please let me know if I'm wrong)
